GPMF_gyro.py

```diff
@@ -74,11 +74,10 @@ def parse_gpmf(self):
         omega = np.array(self.gyro) / self.gyro_scal
         self.num_gyro_samples = omega.shape[0]
 
-
-        self.gyro_rate = self.num_gyro_samples / self.video_length 
+        # gyro data gets written roughly every second
+        self.gyro_rate = self.num_gyro_samples / int(self.video_length)

```

Comment:
I believe this is wrong. Although the gyro data is written roughly every second (like the comment above words it), there is still data about the last fraction of a second. E.g. this assumes that only whole seconds include data...?

I have a 4.2 second long GoPro video, there are 5 chunks of gyro data: 203, 202, 202, 203, and 53 samples respectively. Dividing by the floor-ed value of the video length -- `4` instead of `4.2` -- this makes the gyro_rate way off. The 53 samples are for the last `.2`. If the video was 4.99 seconds, there would be maybe 200 samples in the 5th chunk, but only divided by 4 (`int(4.99)` is `4`).

Actual data from the GoPro file I have:
Before this change: `gyro_rate = 853 / 4.2042 = 205.27091955663383` (correct)
After this change: `gyro_rate = 853 / 4 = 215.75` (wrong)

Reply:
This was in 63a6db7, so I think that commit should be reverted.

Reply:
Which GoPro and firmware are you using?

Reply:
Hero8 Black with Labs firmware

Reply:
Hero 8 has STMP  (timestamp in microseconds) that should produce the best results for estimating the sampling rate. For Hero 6 (fw 2.10, fw 1.6 has TICK which is timestamp in milliseconds) this is the only way to estimate sampling rate as far as I know and yields the correct results as data only gets written every second.

Reply:
Are you saying that Hero 6 does not include the last chunk of gyro? For instance a 4.2 sec video recorded on Hero 6 only has 4 chunks of ~200 samples, and no gyro data for the last 0.2 seconds?